### DISCUSSION TITLE
How to set up on blocking IP on FTP attacks

### DISCUSSION THREAD

**Discussion by waldemarvulf** [DigitalRuby/IPBan#207] [General]
Hello all.
Is it possible to configure the blocking of IP attacking FTP server?
If so, could anyone tell me how to do this?

I apologize if this is a stupid question. I'm not very experienced in this.
Thanks in advance.

**Comment by jjxtra** [comment]
Of course. Anything that writes to a log file on Windows or Linux can be blocked. The windows event viewer can also be used.

**Reply by waldemarvulf** [reply]
If I understood correctly, then this is configured in a configuration file.
But, as I understand it, in this group there is already a record of blocking by event 4625.
 <Group>
        <Source>RDP</Source>
        <Keywords>0x8010000000000000</Keywords>
        <Path>Security</Path>

        
        <FailedLoginThreshold>0</FailedLoginThreshold>

				
		
        <Expressions>
          <Expression>
            <XPath>//EventID</XPath>
            <Regex>^(4625|5152)$</Regex>
          </Expression>
          <Expression>
            <XPath>//Data[@Name='IpAddress' or @Name='Workstation' or @Name='SourceAddress']</XPath>
            <Regex>
              <![CDATA[
                (?<ipaddress>.+)
              ]]>
            </Regex>
          </Expression>
          <Expression>
            <XPath>//Data[@Name='ProcessName']</XPath>
            <Regex>
              <![CDATA[
                (?<source_IIS>c:\\Windows\\System32\\inetsrv\\w3wp.exe)?$
              ]]>
            </Regex>
          </Expression>
        </Expressions>
      </Group>

According to the RDP, the protocol blocking the iPs works perfectly, but does not work on FTP for the same event 4625. I can’t understand what needs to be added, changed in the configuration file.

**Comment by jjxtra** [comment]
Can you post the event viewer xml from your ftp? Please mask out any sensitive info.

**Reply by waldemarvulf** [reply]
I tried it to connect obviously with the wrong login and password to my ftp. For some reason, the IP address is not determined in the even viewer.

<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Microsoft-Windows-Security-Auditing" Guid="{54849625-5478-4994-a5ba-3e3b0328c30d}" />
    <EventID>4625</EventID>
    <Version>0</Version>
    <Level>0</Level>
    <Task>12544</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8010000000000000</Keywords>
    <TimeCreated SystemTime="2022-07-01T08:09:07.4261830Z" />
    <EventRecordID>905382</EventRecordID>
    <Correlation ActivityID="{7486445e-8d07-0001-1a45-8674078dd801}" />
    <Execution ProcessID="824" ThreadID="10988" />
    <Channel>Security</Channel>
    <Computer>Compi8</Computer>
    <Security />
  </System>
  <EventData>
    <Data Name="SubjectUserSid">S-1-5-18</Data>
    <Data Name="SubjectUserName">Compi8$</Data>
    <Data Name="SubjectDomainName">WORKGROUP</Data>
    <Data Name="SubjectLogonId">0x3e7</Data>
    <Data Name="TargetUserSid">S-1-0-0</Data>
    <Data Name="TargetUserName">999</Data>
    <Data Name="TargetDomainName">
    </Data>
    <Data Name="Status">0xc000006d</Data>
    <Data Name="FailureReason">%%2313</Data>
    <Data Name="SubStatus">0xc0000064</Data>
    <Data Name="LogonType">8</Data>
    <Data Name="LogonProcessName">Advapi  </Data>
    <Data Name="AuthenticationPackageName">MICROSOFT_AUTHENTICATION_PACKAGE_V1_0</Data>
    <Data Name="WorkstationName">Compi8</Data>
    <Data Name="TransmittedServices">-</Data>
    <Data Name="LmPackageName">-</Data>
    <Data Name="KeyLength">0</Data>
    <Data Name="ProcessId">0xbf0</Data>
    <Data Name="ProcessName">C:\Windows\System32\svchost.exe</Data>
    <Data Name="IpAddress">-</Data>
    <Data Name="IpPort">-</Data>
  </EventData>
</Event>

**Reply by jjxtra** [reply]
If you can export this as raw xml it will be much easier to decipher

**Reply by jjxtra** [reply]
What ftp program are you using?

**Reply by waldemarvulf** [reply]
I use standard tools windows.
[ftptry.txt](https://github.com/DigitalRuby/IPBan/files/9040376/ftptry.txt)

**Comment by jjxtra** [comment]
The ip logged by your ftp program is `-`. So IPBan cannot determine what ip to ban.

**Reply by waldemarvulf** [reply]
Yes. I said this above that Windows event viewer does not track the iP when trying to connect to FTP. 
Well, then it will not work.
Thank you Jeff.

**Reply by jjxtra** [reply]
Does the ftp software write to a log file perhaps?

**Reply by waldemarvulf** [reply]
Yes. On this path on my computer there is a log files FTP. 
C:\inetpub\logs\LogFiles\FTPSVC2
There is an iPs of the one who tried to connect.
Here are a few lines from the file.
#Software: Microsoft Internet Information Services 10.0
#Version: 1.0
#Date: 2022-07-06 06:33:04
#Fields: date time c-ip cs-username s-ip s-port cs-method cs-uri-stem sc-status sc-win32-status sc-substatus x-session x-fullpath
2022-07-06 06:33:04 34.140.248.32 - 192.168.1.77 21 ControlChannelOpened - - 0 0 c5ffc711-e0b2-42ea-865b-044ae8b37a84 -
2022-07-06 06:33:04 34.140.248.32 - 192.168.1.77 21 ControlChannelClosed - - 0 0 c5ffc711-e0b2-42ea-865b-044ae8b37a84 -
2022-07-06 06:33:04 34.140.248.32 - 192.168.1.77 21 ControlChannelOpened - - 0 0 04068348-08f8-4ff9-b593-406b9721a88f -
2022-07-06 06:33:04 34.140.248.32 - 192.168.1.77 21 USER anonymous 331 0 0 04068348-08f8-4ff9-b593-406b9721a88f -
2022-07-06 06:33:04 34.140.248.32 - 192.168.1.77 21 PASS anonymous@ 530 1326 42 04068348-08f8-4ff9-b593-406b9721a88f -
2022-07-06 06:33:04 34.140.248.32 - 192.168.1.77 21 ControlChannelClosed - - 0 0 04068348-08f8-4ff9-b593-406b9721a88f -

**Reply by jjxtra** [reply]
Does this recipe work?

https://github.com/DigitalRuby/IPBan/blob/master/Recipes/Windows/LogFile/IIS_FTP.xml

**Reply by waldemarvulf** [reply]
I copied this code and paste it into the configuration file, rebooted the computer, but it does not seem to work. I tried to connect many times with incorrect data, does not block.

**Reply by jjxtra** [reply]
Can you email your ipban.config and iis ftp log file to support@ipban.com?

**Reply by waldemarvulf** [reply]
yes, sure. Thank you.

**Reply by waldemarvulf** [reply]
Hello Jeff. I finally tested ip ban for ftp and it works. The only thing is that there is some delay when blocking, i.e. with many incorrect connection attempts, blocking does not occur immediately, but after some time.

However, I want to thank you very much for your time and help.

**Comment by YAG82** [comment]
Good time of the day. I have a similar problem.
Can you tell me how the ipban was eventually made to work with FTP?
